gpt_researcher.py

```
@@ -0,0 +1,701 @@
+            except OutputParserException as e:
+                print(f'OA Checker Parser Error: {e}. || {top_ss_chunks} || {rst}')
+                return False
+            if len(p_rst.FIGURE_CAPTION + p_rst.TABLE_CAPTION) > 0:
+                return p_rst
+            else:
+                return None
+        else:
+            return None
+    except Exception as e:
+
+        raise Exception('Error occurs in github_checker:', e)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)